### PR TITLE
js tools & strict mode

### PR DESCRIPTION
**Pull request by zelmandesign**
The issue with missing data type in props has been fixed. 
Eslint & prettier have been added to the project.

### CODE REVIEW

File: pages/ssg.tsx [hunk 1]
```diff
@@ -12,9 +12,7 @@ const Ssg: NextPage = () => {
       </Head>
 
       <main className={styles.main}>
-        <h1 className={styles.title}>
```
Comment (pdobrowolskivalue) [inline pages/ssg.tsx:15]: SSG - there is a missing functionality  - on thins page please show the same data as SSR:
https://nextjs.org/docs/basic-features/data-fetching/get-static-props

File: pages/csr.tsx [hunk 2]
```diff
@@ -32,25 +32,21 @@ function Csr() {
...
-                    </h2>
-                    <div className={styles.dflexcol}>
-                      <span>{mission.description}</span>
-                    </div>
-                  </div>
-                )
-              })
-            }
-          </>
+// tutaj jest problem nie wiem czemu spaceData wyskakuje jako blad w buildzie i wyskakuje object is possibly null pewnie deklaracja jakas zla
```
Comment (pdobrowolskivalue) [inline pages/csr.tsx:35]: Please change initial state `null` to `[]`, because you can't do null.map()

**Comment by pdobrowolskivalue** [comment]
To fetch data on client side please use useSWR, https://swr.vercel.app/

File: pages/csr.tsx (hunk 2 above)
Comment (zelmandesign) [reply pages/csr.tsx:35]: it's done

File: pages/ssg.tsx (hunk 1 above)
Comment (zelmandesign) [reply pages/ssg.tsx:15]: Sure my bad. It's done.

File: pages/api/hello.ts [hunk 3]
```diff
@@ -5,9 +5,6 @@ type Data = {
   name: string
```
Comment (pdobrowolskivalue) [inline pages/api/hello.ts:5]: in your project you are not using pages/api but you can read about https://nextjs.org/docs/api-routes/introduction

File: pages/csr-swr.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import styles from '../styles/Home.module.css'
+
+const fetcher = async () => {
+  const response = await fetch('https://api.spacexdata.com/v3/missions')
+  const data = await response.json()
+  return data
+}
+
+function CsrSWR() {
+  const { data, error } = useSWR('missions', fetcher)
```
Comment (pdobrowolskivalue) [inline pages/csr-swr.tsx:12]: nice!

File: pages/api/hello.ts (hunk 3 above)
Comment (zelmandesign) [reply pages/api/hello.ts:5]: ok, that's a good idea.

File: pages/csr-swr.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import useSWR from 'swr'
+import Head from 'next/head'
+import styles from '../styles/Home.module.css'
+
+const fetcher = async () => {
+  const response = await fetch('https://api.spacexdata.com/v3/missions')
+  const data = await response.json()
+  return data
+}
```
[the remark this answers is not in the document]
Comment (zelmandesign) [reply pages/csr-swr.tsx:9]: ok

**Pull request merged**